beacon-chain/blockchain/forkchoice/process_block.go

```diff
@@ -486,7 +486,7 @@ func (s *Store) currentSlot() uint64 {
 
 // updates justified check point in store if a better check point is known
 func (s *Store) updateJustifiedCheckpoint() {
-	if helpers.SlotsSinceEpochStarts(s.currentSlot()) == 0 {
+	if !(helpers.SlotsSinceEpochStarts(s.currentSlot()) == 0) {
```

Comment:
```suggestion
	if helpers.SlotsSinceEpochStarts(s.currentSlot()) > 0 {
```
this is more clear imo

Reply:
Wait is this what you want?

Reply:
No, I dont think that's it. Here's the spec reference:
```python
if not (compute_slots_since_epoch_start(current_slot) == 0):
...
```
Only proceed if it's the first slot of the epoch. I'm going to simplify it to
```go
if s.currentSlot() % params.beaconConfig.SlotsPerSlot == 0 {
}
```

Reply:
Updated to
```go
	if s.currentSlot() % params.BeaconConfig().SlotsPerEpoch != 0 {
		return
	}
```

Reply:
Can you add a comment? It's not clear of the intention. Is this "if slot is an epoch start slot, don't update the justified checkpoint"? 

If so, why?

Reply:
The intension is to "only check and update best justified epoch to last justified epoch" at the epoch boundary slot. Think of epoch transition, it only happens once an epoch

Reply:
Added a comment